### ISSUE TITLE
OSL: Add Partio Support.

### ISSUE THREAD

**Issue by dracwyrm** [dracwyrm/gentoo-ebuilds#25]
Partio can be found here: https://github.com/wdas/partio

I had to disable it in the ebuild to stop a warning message., might be nice to have.

**Comment by redchillipadi** [comment]
I have added initial ebuilds which compile partio for use with osl in a new branch. They are still a work in progress as I need to check the python_compat is correct, and add some more use flags to make some of the dependencies optional.

**Comment by dracwyrm** [comment]
I was going to worry about this later after ebuilds are in portage. I need to ask Gokturk how the review is going. Been a couple weeks. He may have forgotten. XD

Thanks for getting the ebuild up. I will look it over when it's uploaded.

**Comment by dracwyrm** [comment]
This is now on the back burner. We will keep this open as a reminder to do this if/when OSL is working with LLVM 3.7 or better.

**Comment by redchillipadi** [comment]
Sorry, was this in both the waiting on OSL and Post Portage Addition milestones? I changed it by accident and seem to have broken it, I can't get it back into both milestones at the same time.

**Comment by dracwyrm** [comment]
@redchillipadi Yeah. No worries. :) I've been watching OSL upstream like a hawk for any sign of LLVM 3.7+ support, but nothing yet.

**Comment by redchillipadi** [comment]
Marsupial just submitted a pull request (#693) for OpenShadingLanauge which may allow it to work with LLVM 3.9! If it works then we may be able to bring back OSL and Partio. I am away for the weekend but will check it out next week.

**Comment by dracwyrm** [comment]
I saw that. It's only compatible with master which has is the 1.8 dev branch. We can test it and if they merge, then wait for the next dev release.

**Comment by redchillipadi** [comment]
It appears to work

I compiled OSL using the pull request from marsupial, and recompiled blender-2.78a with the osl USE flag. I am using sys-devel/llvm-3.9.1-r1. I have put this in a new branch called osl

Blender is now able to render my mandelbrot file from Issue #22 without aborting.

**Comment by dracwyrm** [comment]
@redchillipadi There were several OSL scripts I found, so we will just need to test this. Then once in master branch, we can package it. They are releasing dev tarballs, so we would be able to link to those. We might need to put it in portage as masked and the use flag masked for testing until they do an official release. With Use flag masking, we could still stabilize Blender even through depending on a masked package. :)

**Comment by dracwyrm** [comment]
This is a nice list of OSL scripts:
https://github.com/sambler/myblendercontrib/tree/master/online_mat_lib/material-library/bundled/cycles/scripts

And here:
https://github.com/sambler/osl-shaders

**Comment by dracwyrm** [comment]
I added Partio support to OSL. It's in the main repo. This needs to be tested. Well, first I need to find some partio scripts.
